docs/source/tutorial_worlds.rst

```diff
@@ -0,0 +1,227 @@
+
+
+Hogwild Models
+~~~~~~~~~~~~~~
+For models using hogwild training, the primary concern is to share a thread-safe
+version of the model. This process will vary based on which framework you're
+using, but we'll include a few tips for PyTorch here.
+
+First, check out the best practices here:
+http://pytorch.org/docs/master/notes/multiprocessing.html
+
+The primary things to remember are
+1. call ``model.share_memory()`` and include your model in the ``share()`` function
+2. make sure to switch the multiprocessing start method if CUDA is enabled
+
+
+Batching
+^^^^^^^^
+Batching is set up in the following way (the first step is the same as Hogwild):
+
+1. We set up a starting instance of the world: that is, we use ``create_task``
+   to set up a base world with the appropriate agents and tasks.
+2. We pass this world to a ``BatchWorld``.
+3. We create ``batchsize`` worlds, each initialized from a ``share()``'d
+   version of the world and the agents therein.
+
+Now, every time we call ``parley`` on this BatchWorld, we will complete ``batchsize`` examples.
+Note that this is different than the behavior of HogwildWorld, where only a
+single example is executed for each call to parley.
+
+There's a few more complex steps to actually completing a parley in this world.
+
+1. Call ``parley_init`` on each shared world, if the world has it implemented.
+   Most classes don't need this--we currently only use it for our ``MultiWorld``,
+   which handles the case when one specifies multiple separate tasks to launch
+   (e.g. "-t babi,squad"). This does any pre-parley setup, here choosing which
+   sub-tasks to use in the next parley.
+2. Then, iterate over the number of agents involved in the task. For most tasks,
+   this is just two agents: the teacher (task) and the student (model). For each
+   agent, we do two steps:
+
+   a. Call ``BatchWorld.batch_act()``. This method first checks if the **original**
+      isntance of the agent (not the copies) has a function named ``batch_act``
+      implemented and does not have an attribute ``use_batch_act`` set to ``False``.
+      This function is described more below. If condition is not met,
+      the BatchWorld's ``batch_act`` method iterates through each agent copy in the
+      batch and calls the ``act()`` method for that instance.
+      This is the default behavior in most circumstances, and allows agents to
+      immediately work for batching without any extra work--the batch_act method
+      is merely for improved efficiency.
+   b. Call ``BatchWorld.batch_observe()``. This method goes through every **other**
+      agent, and tries to call the ``observe()`` method on those agents. This gives
+      other agents (usually, just the one other agent) the chance to see the action
+      of the agent whose turn it is to act currently.
+
+Next, we'll look at how teachers and models can take advantage of the setup
+above to improve performance.
+
+
+Batched Teachers
```

Comment:
i would say pytorch data loader stuff is missing here?

Reply:
agreed, I think good to have kurt add that

Reply:
@klshuster